assayist/processor/main_analyzer.py

```diff
@@ -0,0 +1,135 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from assayist.processor.base import Analyzer
+from assayist.processor.logging import log
+
+
+class MainAnalyzer(Analyzer):
+    """
+    Look at the json brew output and write neomodels for the basic items identified.
+
+     * The Component
+     * The direct SourceLocation
+     * The Build itself
+     * Any produced Artifacts (RPMs or otherwise)
+     * Any RPMs used in the buildroot
+     * If it's an image build, an RPMs included in the image
+    """
+
+    buildroot_to_artifact = {}
+
+    def map_buildroot_to_artifact(self, buildroot_id, artifact_id):
+        """Store the mapping in self.buildroot_to_artifact."""
+        if buildroot_id in self.buildroot_to_artifact:
+            self.buildroot_to_artifact[buildroot_id].append(artifact_id)
+        else:
+            self.buildroot_to_artifact[buildroot_id] = [artifact_id]
+
+    def read_and_save_buildroots(self):
+        """Save and link the rpms used in the buildroot for each artifact."""
+        buildroots_info = self.read_metadata_file(self.BUILDROOT_FILE)
+        for buildroot_id in buildroots_info:
+            log.debug("Creating artifacts for buildroot %s", buildroot_id)
+            for rpm_info in buildroots_info[buildroot_id]:
+                rpm = self.get_or_create_rpm_artifact_from_hash(rpm_info)
+                if buildroot_id in self.buildroot_to_artifact:
```

Comment:
Optional: There is a lot of indentation in this method. How about:
```python
if buildroot_id not in self.buildroot_to_artifact:
    continue
for artifact in self.buildroot_to_artifact[buildroot_id]:
    artifact.buildroot_artifacts.connect(rpm)
```

Reply:
When would this if statement be false?

Reply:
Ideally never, a buildroot id not existing would mean the data is malformed. I supposed it's better to throw an exception and stop analysis in that case, so I'll just delete the if condition.